### PR TITLE
Fix offset overflow in plain decoder

### PR DESCRIPTION
**Pull request by eddyxu**
This happens when there was a row group that is larger than 4GB. Plain decoder calculating the chunked offsets using `u32` which causes overflow.

### CODE REVIEW

File: rust/src/encodings/plain.rs [hunk 1]
```diff
@@ -310,6 +307,23 @@ impl<'a> PlainDecoder<'a> {
...
 }
 
+fn make_chunked_requests(
+    indices: &[u32],
+    byte_width: usize,
+    block_size: usize,
+) -> Vec<Range<usize>> {
+    let mut chunked_ranges = vec![];
+    let mut start: usize = 0;
+    for (i, w) in indices.windows(2).enumerate() {
```
Comment (changhiskhan) [inline rust/src/encodings/plain.rs:317]: so this is comparing sequential pairs of the indices but originally it's comparing each index with `start`?

File: rust/src/encodings/plain.rs [hunk 2]
```diff
@@ -325,23 +339,12 @@ impl<'a> Decoder for PlainDecoder<'a> {
             return self.take_boolean(indices).await;
         }
 
-        let block_size = self.reader.prefetch_size() as u32;
```
Comment (changhiskhan) [inline rust/src/encodings/plain.rs:328]: so the integer overflow issue is happening because we're casting to u32?

**Comment by changhiskhan** [comment]
@cemoody this should be the fix. We'll make a new release tmr once this is merged

File: rust/src/io/exec/take.rs [hunk 3]
```diff
@@ -67,10 +67,11 @@ impl Take {
                     let rows = if extra.fields.is_empty() {
                         batch
                     } else {
-                        dataset
```
[the remark this answers is not in the document]
Comment (eddyxu) [reply rust/src/io/exec/take.rs:70]: Oh was adding some profiling here. lemme revert it back.

File: rust/src/encodings/plain.rs (hunk 1 above)
Comment (eddyxu) [reply rust/src/encodings/plain.rs:317]: This is the same logic, but just make it more `rust-style`

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**